driver-async/src/main/com/mongodb/async/client/AggregateIterableImpl.java

```diff
@@ -71,7 +71,8 @@
                           final ReadConcern readConcern, final WriteConcern writeConcern, final OperationExecutor executor,
                           final List<? extends Bson> pipeline, final AggregationLevel aggregationLevel) {
         super(clientSession, executor, readConcern, readPreference);
-        this.operations = new AsyncOperations<TDocument>(namespace, documentClass, readPreference, codecRegistry, writeConcern, false);
+        this.operations = new AsyncOperations<TDocument>(namespace, documentClass, readPreference, codecRegistry, writeConcern,
+                true, true);
```

Comment:
Will retryReads be plumbed through in the constructor?

Reply:
I have opted to use a fluent setter for the Iterables.

Reply:
I'm starting to see that we're inconsistent about when we use constructor params and when we use fluent setters.

Reply:
I have followed the guideline. However, there are now some constructors that contain more than 11 parameters with the addition of the `retryReads` parameter, which generates errors when running `./gradlew driver-sync:checkstyleMain`.